### PR TITLE
Add support for Core-Config

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
> PR #53 got bugged after a force push, opening this Patch via a new PR

Operand NFD Patch https://github.com/kubernetes-sigs/node-feature-discovery/pull/433  changed the way nfd-worker get's arguments, this patch looks to update the operator to the new way of running the nfd-worker 

Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

**Comment by marquiz** [comment]
> PR #53 got bugged after a force push, opening this Patch via a new PR

Curious why didn't you just force push this version to PR #53?

File: Makefile [hunk 1]
```diff
@@ -54,6 +53,7 @@ BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
 BUNDLE_IMG ?= controller-bundle:$(VERSION)
 
 # Image URL to use all building/pushing image targets
+CONTAINER_RUN_CMD ?= docker run -u "`id -u`:`id -g`"
```
Comment (marquiz) [inline Makefile:56]: Why do these makefile changes keep haunting us?

File: config/crd/bases/nfd.kubernetes.io_nodefeaturediscoveries.yaml [hunk 2]
```diff
@@ -5,7 +5,6 @@ kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.4.1
-    api-approved.kubernetes.io: "unapproved, experimental-only"
```
Comment (marquiz) [inline config/crd/bases/nfd.kubernetes.io_nodefeaturediscoveries.yaml:8]: Looks like unrelated change -> separate PR(?)

File: config/rbac/role.yaml [hunk 3]
```diff
@@ -302,18 +302,3 @@ rules:
...
-  - securitycontextconstraints
-  verbs:
-  - create
-  - delete
-  - get
-  - list
-  - patch
-  - update
-  - use
-  - watch
```
Comment (marquiz) [inline config/rbac/role.yaml:318]: Unrelated -> separate PR

File: config/samples/nfd.kubernetes.io_v1_nodefeaturediscovery.yaml [hunk 4]
```diff
@@ -6,10 +6,30 @@ metadata:
 spec:
   operand:
     namespace: node-feature-discovery-operator
-    image: gcr.io/k8s-staging-nfd/node-feature-discovery:master
+    image: gcr.io/k8s-staging-nfd/node-feature-discovery:v0.8.1
```
Comment (marquiz) [inline config/samples/nfd.kubernetes.io_v1_nodefeaturediscovery.yaml:9]: Don't change the image

**Review by marquiz — CHANGES_REQUESTED** [review]

File: config/crd/bases/nfd.kubernetes.io_nodefeaturediscoveries.yaml (hunk 2 above)
Comment (ArangoGutierrez) [reply config/crd/bases/nfd.kubernetes.io_nodefeaturediscoveries.yaml:8]: is not a change... `make manifest` is deleting that line everytime 🤦🏽

File: Makefile [hunk 5]
```diff
@@ -7,7 +7,6 @@ GO_FMT ?= gofmt
 CONTAINER_RUN_CMD ?= docker run -u "`id -u`:`id -g`"
 
 MDL ?= mdl
-
```
Comment (marquiz) [inline Makefile:10]: nit: why modify the makefile?

File: config/rbac/role.yaml [hunk 6]
```diff
@@ -315,5 +315,4 @@ rules:
   - patch
   - update
   - use
-  - watch
-
+  - watch
```
Comment (marquiz) [inline config/rbac/role.yaml:318]: nit: Where does this change come from? A newline at the end of the last line would be good

File: config/crd/bases/nfd.kubernetes.io_nodefeaturediscoveries.yaml (hunk 2 above)
Comment (marquiz) [reply config/crd/bases/nfd.kubernetes.io_nodefeaturediscoveries.yaml]: OK. It's just seems really hard for me to understand what is being autogenerated and what. Could you submit these `make manifests` related changes as a separate PR?

File: config/rbac/role.yaml (hunk 6 above)
Comment (ArangoGutierrez) [reply config/rbac/role.yaml]: fixed

File: Makefile (hunk 5 above)
Comment (ArangoGutierrez) [reply Makefile]: fixed

File: config/rbac/role.yaml (hunk 3 above)
Comment (ArangoGutierrez) [reply config/rbac/role.yaml]: fixed

File: config/crd/bases/nfd.kubernetes.io_nodefeaturediscoveries.yaml (hunk 2 above)
Comment (ArangoGutierrez) [reply config/crd/bases/nfd.kubernetes.io_nodefeaturediscoveries.yaml]: I need to find a way to `make manifest` not to remove that line, but so far I haven't found documentation for it, but sure will be a following PR

Comment (marquiz) [reply config/crd/bases/nfd.kubernetes.io_nodefeaturediscoveries.yaml]: 👍

**Review by marquiz — APPROVED** [review]
Looks good now. Thanks for you perseverance @ArangoGutierrez 😄 
/lgtm

**Comment by ArangoGutierrez** [comment]
> Looks good now. Thanks for you perseverance @ArangoGutierrez 😄 
> /lgtm

I am a fast learner
Thanks Markus!
